### DISCUSSION TITLE
Looking for information on automatic chat assign

### DISCUSSION THREAD

**Discussion by alonsoac** [LiveHelperChat/livehelperchat#1921] [General]
Hi. In this page: https://doc.livehelperchat.com/docs/auto-assignment/#how-to-show-pending-chats-order-from-oldest-to-newest

The title says "How to setup round robin" but the text does not seem to explain this.

There is this link that does not work:

 I also recommend to read round robin implementation for chat's https://docs.google.com/file/d/0B5uaj1QNYoNTVjRnd1ozdmJDYlkwMEl0T0p6NnZLeThXeEww/edit

Is it possible to assign chat to user who has longest time without getting a chat?

**Comment by remdex** [comment]
Please read again article...

https://doc.livehelperchat.com/docs/auto-assignment

Auto assignment assign chat based who had chat first and last... There is many options just read article...

**Comment by remdex** [comment]
Just for clarification chats are distributed evenly for operators so each of them at the end if shift should have server similar amount of chats unless they had filled their maximum amount of chats they can have...

**Comment by alonsoac** [comment]
Hello. Will the auto assign system work in this case?
- chats are very slow to work on so agents have many active chats
- agents do not login all the same time, some login at 8am, then other at 9am , others at 10am. 

I see that at 8am when there is only one agent all the chats from day before start assigning to him. 
I have set max chats to 0 because they can have more than 50 active.  
I did not find a configuration to limit by number of pending chats the user already has.

**Reply by remdex** [reply]
Sorry your writing does not make sense. Limit you set per agent including pending and active chats he can have at a single moment. If it does not work on yours like that. You missing something...

Snippet from code.
```
$condition = '(active_chats + pending_chats)';
                        if ($department->exclude_inactive_chats == 1) {
                            $condition = '((pending_chats + active_chats) - inactive_chats)';
                        }
 ```
